docs/flags.md

```diff
@@ -0,0 +1,206 @@
+config.define_string_list('services', args=True)
+cfg = config.parse()
+print(cfg)
+# If you ran tilt up -- --foo bar fe be kafka
+# cfg would be the python dict:
+{"foo": ["bar"],
+ "services": ["fe", "be", "kafka"],
+}
+```
+
+You can then use the values in the returned dictionary in your Tiltfile code to decide how to configure Tilt. You can make Tilt work for your team's model of options using these option primitives to decide how to call the builtin Tiltfile function you already use to configure youre Tilt.
+
+# (Optional) Set Results
+One very common use for options is which services to run, so we offer special support for that case. The function `config.set_resources` allows you to replace a Tiltfile like:
+```python
+cfg = config.parse()
+if 'fe' in cfg['services']:
+  docker_build(...)
+  k8s_yaml(...)
+  k8s_resource(...)
+if 'be' in cfg['services']:
+  docker_build(...)
+  k8s_yaml(...)
+  k8s_resource(...)
+```
+
+With the simpler:
+```python
+cfg = config.parse()
+config.set_resources(cfg['services'])
+```
+
+## Examples
+
+### Run only some services
+Your app has many services, call them A, B, C, and D. You want users to be able to run only some of them.
+
+and you want your users to be able to run only a subset of them. E.g., `tilt up` will run all services, but running `tilt up a b d` will run only the resources a, b, and d, ignoring c. Each run of Tilt will start from scratch.
+#### Command-Lines
+* `tilt up`: run all services
+* `tilt up a b d`: run A, B, and D but not C
+* `tilt args a b`: change running Tilt to run A and B but neither C nor D.
+* `tilt up` (a second time): run all services
+
+#### Tiltfile
+```python
+config.define_string_list("to-run", args=True)
+cfg = config.parse()
+
+config.set_resources(cfg['to-run'])
+```
+
+### Run a defined set of services
+Your app has many services, call them A, B, C, and D. You have defined modes you want users to be able to start (e.g. consumer vs. enterprise) and you don't want them to have to remember which service is necessary for which mode.
+
+You want to be able to write a doc that tells users to run `tilt up consumer` the first time and know that's enough that they'll be set up. Running Tilt subsequent times without arguments should default to the mode they already have set.
+
+#### Command-Lines
+* `tilt up`: run all services
+* `tilt up consumer`: run consumer services
+
+#### Tiltfile
+```python
+config.define_string_list("to-run", args=True)
+cfg = config.parse()
+
+groups = {
+  'consumer': ['a', 'b', 'c'],
+  'enterprise': ['a', 'b', 'd'],
+}
+
+resources = []
+for arg in cfg['args']:
+  if arg in groups:
+    resources += groups[arg]
+  else:
+    resources.append(r)
+config.set_resources(resources)
+```
+
+### Specify services to edit
+You have many services that your users frequently need to run, but don't expect to edit. Your YAML includes good enough images. (Perhaps they're from a suitably recent CI invocation, or refer to standard images.)
```

Comment:
```suggestion
You have many services that your users frequently need to run, but don't expect to edit. You specify good-enough images in the YAML so your users can just pull them instead of building them. (Perhaps they're from a suitably recent CI invocation, or refer to standard images.) You add a `--to-edit` arg that allows users to specify which images should be built locally.
```